ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -379,20 +380,27 @@ boolean isCaughtUp() {
   }
 
   void shutdown() throws InterruptedException, StoreException, IOException {
-    if (enabled.get()) {
-      logger.info("Hard delete shutdown initiated");
-      enabled.set(false);
-      hardDeleteLock.lock();
-      try {
-        pauseCondition.signal();
-      } finally {
-        hardDeleteLock.unlock();
+    final Timer.Context context = metrics.hardDeleteShutdownTimeInMs.time();
+    try {
+      if (enabled.get()) {
+        logger.info("Hard delete shutdown initiated");
+        enabled.set(false);
+        hardDeleteLock.lock();
+        try {
+          pauseCondition.signal();
+        } finally {
+          hardDeleteLock.unlock();
+        }
+        if (!shutdownLatch.await(5, TimeUnit.SECONDS)) {
```

Comment:
What's the implication of this change? I'm not sure it is ok to just cut off the shutdown since hard delete might be doing actual work @pnarayanan - please comment.